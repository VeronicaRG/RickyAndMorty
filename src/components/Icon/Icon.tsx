import React from 'react';
import { IconProps } from './Icon.types';

export const svgs = {
  item: () => require('@svgs/item.svg'),
  plus: () => require('@svgs/plus.svg'),
  'home-icon': () => require('@svgs/home-icon.svg'),
  'arrow-left': () => require('@svgs/arrow-left.svg'),
  'fav-colorful': () => require('@svgs/fav-colorful.svg'),
  'rick-colorful': () => require('@svgs/rick-colorful.svg'),
  'morty-colorful': () => require('@svgs/morty-colorful.svg'),
  'fav-monochrome': () => require('@svgs/fav-monochrome.svg'),
  'rick-monochrome': () => require('@svgs/rick-monochrome.svg'),
  'morty-monochrome': () => require('@svgs/morty-monochrome.svg'),
} as const;

export const Icon: React.FC<IconProps> = ({ name = 'item', ...rest }) => {
  const Component = svgs[name]();

  return <Component.default color="#fff" height={30} width={30} {...rest} />;
};
