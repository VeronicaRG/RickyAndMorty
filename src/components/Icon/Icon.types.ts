import { svgs } from '.';
import { SvgProps } from 'react-native-svg';

export interface IconProps extends SvgProps {
  name?: keyof typeof svgs;
}
