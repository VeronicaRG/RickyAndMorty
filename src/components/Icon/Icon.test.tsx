import React from 'react';
import { render } from '@testing-library/react-native';
import { Icon, svgs } from './Icon';

describe('Component: Icon', () => {
  it('should render component', () => {
    const { toJSON } = render(
      <>
        {Object.keys(svgs).map(iconName => (
          <Icon key={iconName} name={iconName as keyof typeof svgs} />
        ))}
      </>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
