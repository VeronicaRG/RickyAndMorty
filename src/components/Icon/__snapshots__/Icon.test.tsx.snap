// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Icon should render component 1`] = `
[
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
  <SvgMock
    color="#fff"
    height={30}
    width={30}
  />,
]
`;
